src/lib/useWalletConnect.ts

```diff
@@ -81,17 +81,30 @@ export const useWalletConnect = () => {
       return;
     }
 
-    const authToken = await getAuthToken({
-      address,
-      connector,
-      walletType: WALLET_TYPES.WALLET_CONNECT,
-    });
+    let authToken = Nothing();
+    try {
+      const token = await getAuthToken({
+        address,
+        connector,
+        walletType: WALLET_TYPES.WALLET_CONNECT,
+      });
+      authToken = Just(token);
+    } catch (e) {
+      if (e.message === 'METHOD_NOT_SUPPORTED') {
+        console.warn(
+          'wallet does not support message signing. proceeding without auth token.'
+        );
+      } else {
+        throw e;
+        //TODO  should errors with this *really* prevent login?
```

Comment:
This is a good point, do we actually even need the auth token for any connected WC wallet? If the user is pointing their device at a QR code, or pasting the WC URL into another wallet, is it safe to assume that they also want to "authenticate"? In other words, by connecting, one is also implicitly authenticating, before actually clicking Authenticate.

To avoid refactoring, maybe we can keep the existing Authenticate flow for wallets that do support `signMessage`, and when hitting this error branch, just login anyways. It's actually a nice feature - one less popup prompt on the connected peer :)

Reply:
> do we actually even need the auth token for any connected WC wallet?

Reminder that we actually get these auth tokens for a reason!

There's a couple places we want to be able to deterministically derive new secrets. When setting networking keys, or generating invite codes for example. For custodial wallets, where we have access to a seed or private key, we can just derive from that. For non-custodial wallets, we do not have access to any secrets, so we cannot create any new (deterministic) secrets. This is why we sign an "auth token" on login: it gives us a secret to work with.

tldr: we can't just toss is out without facing ux consequences.

> maybe we can keep the existing Authenticate flow for wallets that do support signMessage, and when hitting this error branch, just login anyways.

Isn't this what we do as of e1b0301? We catch the "not supported" case and login anyway. It's _other_ errors, unexpected ones, that I'm wondering about here. For example, if we're dumb and mess up the function call, or their wallet implements it poorly, and we get a "invalid argument" error or w/e, should that prevent login? We probably don't want to _silently_ proceed, because it implies changed UX as per the above...

Reply:
Thanks for the explanation, this is much clearer now.

In this error case, since it could have failed for a variety of known and unknown reasons, and since we need an auth token for deterministically deriving new secrets, perhaps this `throw` is actually the best choice for now - it will render the message asking the user to file an issue, and then we can eventually add support for the failing WalletConnect peers.